tests/tests/fixtures/querygen/groupbygen.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023-2025 ParadeDB, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use proptest::prelude::*;
+use std::fmt::Debug;
+
+/// Represents an item in the SELECT list
+#[derive(Clone, Debug, PartialEq)]
+pub enum SelectItem {
+    Column(String),
+    Aggregate(String),
+}
+
+/// Represents a GROUP BY expression with an explicit target list
+#[derive(Clone, Debug)]
+pub struct GroupByExpr {
+    pub group_by_columns: Vec<String>,
+    pub target_list: Vec<SelectItem>,
+}
+
+impl GroupByExpr {
+    pub fn to_sql(&self) -> String {
+        if self.group_by_columns.is_empty() {
+            String::new()
+        } else {
+            format!("GROUP BY {}", self.group_by_columns.join(", "))
+        }
+    }
+
+    pub fn to_select_list(&self) -> String {
+        self.target_list
+            .iter()
+            .map(|item| match item {
+                SelectItem::Column(col) => col.clone(),
+                SelectItem::Aggregate(agg) => agg.clone(),
+            })
+            .collect::<Vec<_>>()
+            .join(", ")
+    }
+}
+
+/// Generate arbitrary GROUP BY expressions with random target list ordering
+pub fn arb_group_by(
+    columns: Vec<impl AsRef<str>>,
+    aggregates: Vec<&'static str>,
+) -> impl Strategy<Value = GroupByExpr> {
+    let columns = columns
+        .into_iter()
+        .map(|c| c.as_ref().to_string())
+        .collect::<Vec<_>>();
+
+    // Generate 0-3 grouping columns from the available columns
+    (0..=3.min(columns.len())).prop_flat_map(move |group_size| {
```

Comment:
You can directly select between 0 and 3 random columns from the list with:
```suggestion
    // Generate 0-3 grouping columns from the available columns
    proptest::sample::subsequence(columns, 0..3).prop_flat_map(|selected_columns| {
```